packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -1510,20 +1541,18 @@ export class CallContext {
   }
 
   public setLatestNotification(callId: string, notification: CallNotification): void {
-    this._latestCallIdsThatPushedNotifications[notification.target] = callId;
+    this._latestCallIdOfNotification[notification.target] = callId;
     this.modifyState((draft: CallClientState) => {
       draft.latestNotifications[notification.target] = notification;
     });
   }
 
-  public deleteLatestNotification(callId: string, notificationTarget: NotificationTarget): void {
-    let callIdToPushLatestNotification = this._latestCallIdsThatPushedNotifications[notificationTarget];
-    callIdToPushLatestNotification = callIdToPushLatestNotification
-      ? this._callIdHistory.latestCallId(callIdToPushLatestNotification)
-      : undefined;
-    // Only delete the notification if the call that pushed the notification is the same as the call that is trying
-    // to delete it.
-    if (callIdToPushLatestNotification !== callId) {
+  public deleteLatestNotification(callId: string | undefined, notificationTarget: NotificationTarget): void {
```

Comment:
I will refactor this function in a follow-up PR to change the optional callId to be the 2nd argument

Reply:
This will be done before release time?

Reply:
Yes. Just did not want to make this PR too big